File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Python/s580342216_comments_removed.py
import sys

class Solution:
    def __init__(self):
        for line in sys.stdin:
            self.run_case(line.strip())
    
    def run_case(self, line):
        parts = line.split()
        N = int(parts[0])
        K = int(parts[1])
        length = 0
        while N > 0:
            N = N // K
            length +=1
        print(length)

class LCS:
    def lcs(self, A, B):
        m, n = len(A), len(B)
        self.dp = [[0]*(n+1) for _ in range(m+1)]
        for i in range(1, m+1):
            for j in range(1, n+1):
                if A[i-1] == B[j-1]:
                    self.dp[i][j] = self.dp[i-1][j-1] +1
                else:
                    self.dp[i][j] = max(self.dp[i-1][j], self.dp[i][j-1])
        return self.dp[m][n]
    
    def getLCS(self, X, Y):
        m, n = len(X), len(Y)
        index = self.dp[m][n]
        temp = index
        lcs = [''] * (index + 1)
        i, j = m, n
        while i > 0 and j > 0:
            if X[i-1] == Y[j-1]:
                lcs[index-1] = X[i-1]
                i -=1
                j -=1
                index -=1
            elif self.dp[i-1][j] > self.dp[i][j-1]:
                i -=1
            else:
                j -=1
        return ''.join(lcs[:temp])

if __name__ == "__main__":
    Solution()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/Java/Go/s580342216_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type Solution struct {
	scanner *bufio.Scanner
}

func main() {
	solution := Solution{scanner: bufio.NewScanner(os.Stdin)}
	for solution.scanner.Scan() {
		solution.run_case()
	}
}

func (s *Solution) run_case() {
	line := s.scanner.Text()
	parts := strings.Fields(line)
	N, _ := strconv.Atoi(parts[0])
	K, _ := strconv.Atoi(parts[1])

	length := 0
	for N > 0 {
		N /= K
		length++
	}

	fmt.Println(length)
}

func strToIntArray(str string) []int {
	vals := strings.Fields(str)
	sz := len(vals)
	res := make([]int, sz)
	for i := 0; i < sz; i++ {
		res[i], _ = strconv.Atoi(vals[i])
	}
	return res
}

type LCS struct {
	dp [][]int
}

func (l *LCS) lcs(A, B string) int {
	szA, szB := len(A), len(B)
	l.dp = make([][]int, szA+1)
	for i := range l.dp {
		l.dp[i] = make([]int, szB+1)
	}

	for i := 0; i <= szA; i++ {
		l.dp[i][0] = 0
	}
	for j := 0; j <= szB; j++ {
		l.dp[0][j] = 0
	}

	for i := 1; i <= szA; i++ {
		for j := 1; j <= szB; j++ {
			if A[i-1] == B[j-1] {
				l.dp[i][j] = l.dp[i-1][j-1] + 1
			} else {
				l.dp[i][j] = max(l.dp[i-1][j], l.dp[i][j-1])
			}
		}
	}

	return l.dp[szA][szB]
}

func (l *LCS) getLCS(X, Y string) string {
	m, n := len(X), len(Y)

	index := l.dp[m][n]
	temp := index

	lcs := make([]byte, index)
	i, j := m, n
	for i > 0 && j > 0 {
		if X[i-1] == Y[j-1] {
			lcs[index-1] = X[i-1]
			i--
			j--
			index--
		} else if l.dp[i-1][j] > l.dp[i][j-1] {
			i--
		} else {
			j--
		}
	}

	return string(lcs[:temp])
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

// End of Code
